app/dao/services_dao.py

```diff
@@ -403,6 +404,8 @@ def _delete_commit(query):
     list(map(db.session.delete, verify_codes))
     db.session.commit()
     users = [x for x in service.users]
+    for user in users:
+        _delete_commit(TemplateCategory.query.filter_by(created_by_id=user.id))
```

Comment:
What happens if a TC created by this user is still in use?

Reply:
As it stands currently if the category that a user created is in use by any templates that user / user's service does not own then this delete will fail with a foreign key constraint violation:`fk_template_template_categories`. 

So to delete a user we'd need to delete their TC which doesn't make sense if it's in use by other templates. If we remove the FK constraint between the user <-> TC we end up with TC's referencing users that may not exist. Technically that should be fine as we typically just archive a user rather than delete.